src/angular-app/languageforge/lexicon/editor/editor.component.ts

```diff
@@ -87,26 +87,40 @@ export class LexiconEditorController implements angular.IController {
   ];
 
   constructor(private readonly $filter: angular.IFilterService,
-              private readonly $interval: angular.IIntervalService,
-              private readonly $q: angular.IQService,
-              private readonly $scope: angular.IScope,
-              private readonly $state: angular.ui.IStateService,
-              private readonly $window: angular.IWindowService,
-              private readonly activityService: ActivityService,
-              private readonly applicationHeaderService: ApplicationHeaderService,
-              private readonly modal: ModalService,
-              private readonly notice: NoticeService,
-              private readonly sessionService: SessionService,
-              private readonly semanticDomains: SemanticDomainsService,
-              private readonly commentService: LexiconCommentService,
-              private readonly editorService: EditorDataService,
-              private readonly lexService: LexiconEntryApiService,
-              private readonly lexProjectService: LexiconProjectService,
-              private readonly rightsService: LexiconRightsService,
-              private readonly sendReceive: LexiconSendReceiveService,
-             ) {}
+    private readonly $interval: angular.IIntervalService,
+    private readonly $q: angular.IQService,
+    private readonly $scope: angular.IScope,
+    private readonly $state: angular.ui.IStateService,
+    private readonly $window: angular.IWindowService,
+    private readonly activityService: ActivityService,
+    private readonly applicationHeaderService: ApplicationHeaderService,
+    private readonly modal: ModalService,
+    private readonly notice: NoticeService,
+    private readonly sessionService: SessionService,
+    private readonly semanticDomains: SemanticDomainsService,
+    private readonly commentService: LexiconCommentService,
+    private readonly editorService: EditorDataService,
+    private readonly lexService: LexiconEntryApiService,
+    private readonly lexProjectService: LexiconProjectService,
+    private readonly rightsService: LexiconRightsService,
+    private readonly sendReceive: LexiconSendReceiveService,
+  ) { }
 
   $onInit(): void {
+    angular.element(window).bind('keyup', (e: Event) => {
```

Comment:
I'm wondering whether `onInit` is the right lifecycle hook to set this up or not?  If that's convention, then ok, however I would probably be more inclined to set the listener up after the data is back and we actually realize we have a need for the ability to move around.  Additionally, will this clean-up the listeners after the user leaves this page?

Reply:
In Angularjs onInit will occur after all elements have been constructed and bound.

Reply:
Per our discussion, binding to `window` is acceptable.  However, we need to unregister the event listener on the $onDestroy() lifecycle hook.

Reply:
There are console errors that need to be addressed before taking a look at this again.

The console errors have to do with calling the method in a context that it doesn't expect.

I will approve with the current approach of $onInit / $onDestroy

However, an alternative approach could be to use the $on('$locationChangeStart') with an eye to registering the handler when the location is changed to /editor/entry and unregister the handler when the location is changed to something else.

Reply:
> However, an alternative approach could be to use the $on('$locationChangeStart') with an eye to registering the handler when the location is changed to /editor/entry and unregister the handler when the location is changed to something else.

💡  I get it now...this is a trip wire for the editor where pg-up/pg-dn are relevant.  It's different than relying on the presence/absence of a particular DOM element.  If I'm understanding this properly, this does sound perfect!